In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import roc_auc_score, average_precision_score

df = pd.read_csv(r"C:\Users\Asus\Documents\prj\Fraud-GNN-Detection\data\creditcard.csv")

X = df.drop("Class", axis=1)
y = df["Class"]

scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)

X_train, X_test, y_train, y_test = train_test_split(
    X_scaled, y, test_size=0.2, stratify=y, random_state=42
)

model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)

probs = model.predict_proba(X_test)[:, 1]

roc = roc_auc_score(y_test, probs)
pr = average_precision_score(y_test, probs)

print("Logistic Regression ROC-AUC:", roc)
print("Logistic Regression PR-AUC:", pr)

Logistic Regression ROC-AUC: 0.9560341968807783
Logistic Regression PR-AUC: 0.7438200889614668


This is a very strong baseline — seriously good.

📊 Your Result
Model	ROC-AUC	PR-AUC
Logistic Regression	0.956	0.744

1️⃣ Random Forest

In [2]:
from sklearn.ensemble import RandomForestClassifier

rf = RandomForestClassifier(n_estimators=100, n_jobs=-1)
rf.fit(X_train, y_train)

probs = rf.predict_proba(X_test)[:, 1]

roc = roc_auc_score(y_test, probs)
pr = average_precision_score(y_test, probs)

print("Random Forest ROC-AUC:", roc)
print("Random Forest PR-AUC:", pr)

Random Forest ROC-AUC: 0.9576946211799294
Random Forest PR-AUC: 0.8636754161115049


2️⃣ XGBoost (Very Important for Paper)

In [3]:

from xgboost import XGBClassifier

xgb = XGBClassifier(
    n_estimators=200,
    max_depth=6,
    scale_pos_weight=(y_train == 0).sum() / (y_train == 1).sum()
)

xgb.fit(X_train, y_train)

probs = xgb.predict_proba(X_test)[:, 1]

roc = roc_auc_score(y_test, probs)
pr = average_precision_score(y_test, probs)

print("XGBoost ROC-AUC:", roc)
print("XGBoost PR-AUC:", pr)

XGBoost ROC-AUC: 0.9654549917884994
XGBoost PR-AUC: 0.880381640438991
